src/replay/execution.zig

```diff
@@ -37,14 +39,42 @@ pub const ReplayExecutionState = struct {
     slot_tracker: *SlotTracker,
     epochs: *EpochTracker,
     progress_map: *ProgressMap,
-    status_cache: *sig.core.StatusCache,
+
+    // owned
+    status_cache: sig.core.StatusCache,
+
+    pub fn init(
+        allocator: Allocator,
+        logger: sig.trace.Logger,
+        my_identity: Pubkey,
+        thread_pool: *ThreadPool,
+        account_store: AccountStore,
+        ledger_reader: *LedgerReader,
+        slot_tracker: *SlotTracker,
+        epochs: *EpochTracker,
+        progress_map: *ProgressMap,
+    ) Allocator.Error!ReplayExecutionState {
+        return .{
+            .allocator = allocator,
+            .logger = .from(logger),
+            .my_identity = my_identity,
+            .vote_account = null, // voting not currently supported
+            .account_store = account_store,
+            .thread_pool = thread_pool,
+            .ledger_reader = ledger_reader,
+            .slot_tracker = slot_tracker,
+            .epochs = epochs,
+            .progress_map = progress_map,
+            .status_cache = .DEFAULT,
+        };
+    }
 };
 
 /// 1. Replays transactions from all the slots that need to be replayed.
 /// 2. Store the replay results into the relevant data structures.
 ///
 /// Analogous to [replay_active_banks](https://github.com/anza-xyz/agave/blob/3f68568060fd06f2d561ad79e8d8eb5c5136815a/core/src/replay_stage.rs#L3356)
-pub fn replayActiveSlots(state: ReplayExecutionState) !bool {
+pub fn replayActiveSlots(state: *ReplayExecutionState) !bool {
```

Comment:
Related to the other comment on ReplayExecutionState, it doesn't own anything, why is this now a mutable pointer?

Reply:
[bf4581c](https://github.com/Syndica/sig/pull/810/commits/bf4581c3d3cfd4de4091d277c59b7ff16f474fcd)